src/libguac/guacamole/client-fntypes.h

```diff
@@ -79,5 +91,22 @@ typedef void guac_client_log_handler(guac_client* client,
  */
 typedef int guac_client_init_handler(guac_client* client);
 
+/**
+ * A function that will broadcast arbitrary data to a subset of users for
+ * the provided client, using the provided user callback for any user-specific
+ * operations.
+ *
+ * @param client
+ *     The guac_client associated with the users to broadcast to.
+ *
+ * @param callback
+ *     A callback that should be invoked with each broadcasted user.
+ *
+ * @param data
+ *     Arbitrary data that may be used to broadcast to the subset of users.
+ */
+typedef void guac_client_broadcast_handler(
+        guac_client* client, guac_user_callback* callback, void* data);
```

Comment:
This particular function type ...

* ... is specific to `guac_socket`, and should instead be named `guac_socket_broadcast_handler` within files specific to `guac_socket`
* ... is internal to the broadcast socket implementation and need not be exposed publicly. It should instead be declared strictly internally and omitted from the public API.